cqc/pythonLib.py

```diff
@@ -1593,15 +1929,46 @@ def check_active(self):
         """
         Checks if the qubit is active
         """
-        if self._cqc.pend_messages:
-            return  # will be handled in the flush, not here
         if not self._active:
-            raise QubitNotActiveError("Qubit is not active, has either been sent, measured, released or not received")
+
+            # This conditional checks whether it is certain that the qubit is inactive at this 
+            # point in the code. If such is the case, an error is raised. 
+            # At this point, it is certain that self_active is False. However, this does not necessarily
+            # mean that the qubit is inactive due to the possibility to write cqc_if blocks.
+            # There are four options:
+            # 1) Control is currently not inside a CQCMix. In that case, the qubit is inactive.
+            # 2) The qubit was deactivated in the current scope. The qubit therefore is inactive.
+            # 3) The qubit was deactivated in an ancestor scope. The qubit therefore is inactive.
+            # 4) The qubit was deactivated in a descendent scope.  The qubit is therefore inactive. 
+            # The only possible way self_active can be False but the qubit is in fact active, is
+            # if the qubit was deactivated in a sibling scope, such as the sibling if-block of an else-block.
+            if (
+                not self._cqc._inside_cqc_mix
+                or self.scope_of_deactivation == self._cqc.current_scope
+                or self.scope_of_deactivation in self._cqc.current_scope.ancestors
+                or self.scope_of_deactivation in self._cqc.current_scope.descendants
+            ):
+
+                raise QubitNotActiveError("""
+        Qubit is not active. Possible causes:
```

Comment:
Nit: This indentation looks strange to me, maybe do
```python
                raise QubitNotActiveError(
                    " Qubit is not active. Possible causes:"
                    "- Qubit is sent to another node"
                    "- Qubit is measured (with inplace=False)"
                    "- Qubit is released"
                    "- Qubit is not received"
                    "- Qubit is used and created in the same factory"
                    "- Qubit is measured (with inplace=False) inside a cqc_if block earlier in the code"
                )
```

Reply:
Done